src/client/game.cpp

```diff
@@ -2439,6 +2446,17 @@ void Game::updateCameraOrientation(CameraOrientation *cam, float dtime)
 }
 
 
+// Get the state of an optionally togglable key
+bool Game::getKeyOrToggleState(const GameKeyType key, const bool should_key_toggle, const bool current_key_state)
```

Comment:
I feel like this could be less verbose, e.g.

```suggestion
bool Game::getToggleKeyState(GameKeyType key, bool toggling_enabled, bool prev_state)
```

(the const doesn't have a meaningful effect here as far as I can see)
(`toggling_enabled` for the lack of a better idea, `should_toggle` sounds like "flip the toggle now if this value is true")

Reply:
Fair, I was thinking it helps with explicitly communicating that e.g. current_key_state wouldn't be set by the function itself, but renaming it to previous state clears that up nicely. The other name suggestions are also solid, adapting them a bit to keep things explicit.